adapter/internal/messaging/notification_listener.go

```diff
@@ -318,14 +323,13 @@ func handleSubscriptionEvents(data []byte, eventType string) {
 		return
 	}
 	if subscriptionEvent.Event.Type == subscriptionCreate {
-		updateSubscription(subscriptionEvent.SubscriptionID, sub)
+		eh.SubscriptionMap[sub.SubscriptionID] = &sub
 	} else if subscriptionEvent.Event.Type == subscriptionUpdate {
-		eh.SubList.List = removeSubscription(eh.SubList.List, subscriptionEvent.SubscriptionID)
-		updateSubscription(subscriptionEvent.SubscriptionID, sub)
+		eh.SubscriptionMap[sub.SubscriptionID] = &sub
 	} else if subscriptionEvent.Event.Type == subscriptionDelete {
-		eh.SubList.List = removeSubscription(eh.SubList.List, subscriptionEvent.SubscriptionID)
+		delete(eh.SubscriptionMap, sub.SubscriptionID)
 	}
```

Comment:
How about adding logs for other events as well? same, that have already done for the application events.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/2126/commits/618511c91e7788869966c069893c6ec4acb27919